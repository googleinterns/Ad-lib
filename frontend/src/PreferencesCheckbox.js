import React from 'react';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';

export default function PreferencesCheckbox(props) {
  return (
    <FormControlLabel
      label="Save my preferences for later"
      labelPlacement="start"
      control={
        <Checkbox
          onChange={(event) => props.onChange(event.target.checked)}
          name="checkbox"
          color="primary"
          defaultChecked="true"
        />
      }
    />
  );
}
