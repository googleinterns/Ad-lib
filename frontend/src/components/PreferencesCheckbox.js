import React from 'react';
import PropTypes from 'prop-types';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';

// Add onChange to props validation
PreferencesCheckbox.propTypes = {
  onChange: PropTypes.func,
};

/**
 * Create checkbox component for input on saving preferences for future use
 * @return {PreferencesCheckbox} PreferencesCheckbox component
 * @param {Object} props
 */
export default function PreferencesCheckbox(props) {
  return (
    <FormControlLabel
      label="Save my preferences for later"
      labelPlacement="start"
      control={
        <Checkbox
          onChange={(event) => props.onChange(event.target.checked)}
          name="checkbox"
          color="primary"
          defaultChecked="true"
        />
      }
    />
  );
}
