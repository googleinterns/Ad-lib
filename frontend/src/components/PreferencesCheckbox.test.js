import React from 'react';
import PreferencesCheckbox from './PreferencesCheckbox';
import renderer from 'react-test-renderer';


describe('Preferences Checkbox', () => {
  it('should be defined', () => {
    expect(PreferencesCheckbox).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = renderer.create(<PreferencesCheckbox />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});