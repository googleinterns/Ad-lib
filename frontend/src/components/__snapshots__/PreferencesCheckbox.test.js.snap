// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preferences Checkbox should render correctly 1`] = `
<label
  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
>
  <span
    aria-disabled={false}
    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-2 Mui-checked MuiIconButton-colorPrimary"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={null}
  >
    <span
      className="MuiIconButton-label"
    >
      <input
        className="PrivateSwitchBase-input-4"
        data-indeterminate={false}
        defaultChecked={true}
        name="checkbox"
        onChange={[Function]}
        type="checkbox"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </span>
  </span>
  <span
    className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
  >
    Save my preferences for later
  </span>
</label>
`;
